src/debugger.ts

```diff
@@ -0,0 +1,245 @@
+        program: "ruby ${file}",
+        env: this.ruby.env,
+      },
+      {
+        type: "ruby_lsp",
+        name: "Debug",
+        request: "launch",
+        // eslint-disable-next-line no-template-curly-in-string
+        program: "ruby -Itest ${relativeFile}",
+        env: this.ruby.env,
+      },
+      {
+        type: "ruby_lsp",
+        name: "Debug",
+        request: "attach",
+        env: this.ruby.env,
+      },
+    ];
+  }
+
+  // Resolve the user's debugger configuration. Here we receive what is configured in launch.json and can modify and
+  // insert defaults for the user. The most important thing is making sure the Ruby environment is a part of it so that
+  // we launch using the right bundle and Ruby version
+  resolveDebugConfiguration?(
+    _folder: WorkspaceFolder | undefined,
+    debugConfiguration: DebugConfiguration,
+    _token?: CancellationToken
+  ): ProviderResult<DebugConfiguration> {
+    // If the user has their own debug launch configurations, we still need to inject the Ruby environment
+    debugConfiguration.env = this.ruby.env;
+    return debugConfiguration;
+  }
+
+  dispose() {
+    if (this.debugProcess) {
+      this.debugProcess.kill("SIGTERM");
+    }
+
+    this.subscriptions.forEach((subscription) => subscription.dispose());
+  }
+
+  private attachDebuggee(): Promise<DebugAdapterDescriptor | undefined> {
+    // When using attach, a process will be launched using Ruby debug and it will create a socket automatically. We have
+    // to find the available sockets and ask the user which one they want to attach to
+    const socketsDir = path.join("/", "tmp", "ruby-lsp-debug-sockets");
+    const sockets = fs
+      .readdirSync(socketsDir)
+      .map((file) => file)
+      .filter((file) => file.endsWith(".sock"));
+
+    return new Promise((resolve, reject) => {
+      if (sockets.length === 0) {
+        reject(new Error("No debuggee processes found"));
+      }
+
+      return window
+        .showQuickPick(sockets, {
+          placeHolder: "Select a debuggee",
+          ignoreFocusOut: true,
+        })
+        .then((selectedSocket) => {
+          if (selectedSocket === undefined) {
+            reject(new Error("No debuggee selected"));
+          } else {
+            resolve(
+              new DebugAdapterNamedPipeServer(
+                path.join(socketsDir, selectedSocket)
+              )
+            );
+          }
+        });
+    });
+  }
+
+  private spawnDebuggeeForLaunch(
+    session: DebugSession
+  ): Promise<DebugAdapterDescriptor | undefined> {
+    let initialMessage = "";
+    const sockPath = this.socketPath();
+
+    return new Promise((resolve, reject) => {
+      this.debugProcess = spawn(
+        "bundle",
+        [
+          "exec",
+          "rdbg",
+          "--open",
+          "--command",
+          `--sock-path=${sockPath}`,
+          "--",
+          session.configuration.program,
```

Comment:
Do we want to add a `useBundler` config like `vscode-rdbg` does? When set to true, it prepends `bundle exec` to the program.

Reply:
I don't think it's necessary. In the configurations, you can just use
```json
{
  "program": "bundle exec ruby ${file}"
}
```
I think it's simpler than introducing another field just to concatenate `bundle exec` for the user.

Reply:
Yeah sure let's keep it simple first.